portal/worker/src/walrus-sites-sw.ts

```diff
@@ -5,13 +5,28 @@ import { getDomain, getSubdomainAndPath } from "@lib/domain_parsing";
 import { redirectToAggregatorUrlResponse, redirectToPortalURLResponse } from "@lib/redirects";
 import { getBlobIdLink, getObjectIdLink } from "@lib/links";
 import resolveWithCache from "./caching";
-import { resolveAndFetchPage } from "@lib/page_fetching";
+import { PageFetcher } from "@lib/page_fetching";
+import { ResourceFetcher } from "@lib/resource";
+import { RPCSelector } from "@lib/rpc_selector";
+import { SuiNSResolver } from "@lib/suins";
+import { WalrusSitesRouter } from "@lib/routing";
 
 // This is to get TypeScript to recognize `clients` and `self` Default type of `self` is
 // `WorkerGlobalScope & typeof globalThis` https://github.com/microsoft/TypeScript/issues/14877
 declare var self: ServiceWorkerGlobalScope;
 declare var clients: Clients;
 
+const rpcUrlList = process.env.RPC_URL_LIST;
+if (!rpcUrlList) {
+    throw new Error("Missing RPC_URL_LIST environment variable");
+}
+const rpcSelector = new RPCSelector(rpcUrlList.split(','));
+export const pageFetcher = new PageFetcher(
```

Comment:
Question: same here, do we need the singleton?

Reply:
Let's track this conversation inside the thread above to avoid duplicate comments: https://github.com/MystenLabs/walrus-sites/pull/326#discussion_r1889032959